### PR TITLE
PUT/GET/DELETE lifecycle configurations

### PR DESCRIPTION
**Pull request by yuejuSDS**
[https://github.com/opensds/multi-cloud/issues/694](https://github.com/opensds/multi-cloud/issues/694)

change:
1.PUT/GET/DELETE lifecycle rules, API should be compatible with AWS S3. Synchronously modify the sending parameters
2.Registered backend support YIG
捕获

@wisererik @anvithks please review and merge.
yig
BUCKET

### CODE REVIEW

**Comment by anvithks** [comment]
@jimccfun Please describe what this PR is doing in the description. It is difficult to review without any context.

**Comment by anvithks** [comment]
@jimccfun Please create an issue in the Dasboard repository for this PR so that we can track the overall changes.

**Comment by yuejuSDS** [comment]
@anvithks  This PR is associated with the issue of multi-cloud, and I don't think it is necessary to create another issue in dashboard

**Comment by anvithks** [comment]
@jimccfun We are creating issues in all the repositories to track progress of work. For any future requirements please create issues. Even if there are issues created in the multi-cloud repo we can link them to these issues and refer them in PRs.

File: src/app/shared/utils/consts.ts [hunk 1]
```diff
@@ -50,6 +50,7 @@ export const Consts = {
         'fusionstorage-object': "FusionStorage Object",
         'ceph-s3': "Ceph S3",
         'gcp-s3': "GCP Storage",
-        'ibm-cos': "IBM COS"
+        'ibm-cos': "IBM COS",
+        'yig': "YIG"
```
Comment (anvithks) [inline src/app/shared/utils/consts.ts]: Have you uploaded a logo for the YIG backend?

File: src/app/business/home/home.component.ts [hunk 2]
```diff
@@ -280,7 +291,8 @@ export class HomeComponent implements OnInit {
             this.selectedType = null;
             let fs_arr = this.Allbackends['fusionstorage-object'] ? this.Allbackends['fusionstorage-object'] : [];
             let ceph_arr = this.Allbackends['ceph-s3'] ? this.Allbackends['ceph-s3'] : [];
-            this.typeDetail = fs_arr.concat(ceph_arr);
+            let yig_arr = this.Allbackends['yig'] ? this.Allbackends['yig'] : [];
+            this.typeDetail = fs_arr.concat(ceph_arr,yig_arr);
```
Comment (anvithks) [inline src/app/business/home/home.component.ts]: Why yig and ceph backends are concat together?

**Comment by anvithks** [comment]
@jimccfun IPlease add more screenshots of the pages where the changes are made and how the screens look like after adding a YIG Backend.

File: src/app/shared/utils/consts.ts (hunk 1 above)
Comment (yuejuSDS) [reply src/app/shared/utils/consts.ts]: The logo of YIG has not been confirmed yet, and it will be uploaded after confirmation later

**Comment by yuejuSDS** [comment]
@anvithks In this PR I have linked the issue that was created in multi-cloud, so do I need to create another issue in dashboard?

File: src/app/business/home/home.component.ts (hunk 2 above)
Comment (yuejuSDS) [reply src/app/business/home/home.component.ts]: Since both ceph_arr and yig_arr need to concat into fs_arr, we can concat at the same time.

**Comment by anvithks** [comment]
> @anvithks In this PR I have linked the issue that was created in multi-cloud, so do I need to create another issue in dashboard?

Yes we are tracking issues in the repository as well.

File: src/app/business/home/home.component.ts (hunk 2 above)
Comment (anvithks) [reply src/app/business/home/home.component.ts]: Ok. I see it now.

File: src/app/shared/utils/consts.ts (hunk 1 above)
Comment (anvithks) [reply src/app/shared/utils/consts.ts]: Ok. Create an issue for this so that we can track it later.

**Comment by anvithks** [comment]
@nguptaopensds Please review

**Comment by anvithks** [comment]
@kumarashit @wisererik please review and approve

**Pull request merged**